src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -824,6 +816,18 @@ private void initChannelPipeline(ChannelPipeline pipeline) {
             aggregateContentForFiltering(pipeline, numberOfBytesToBuffer);
         }
 
+        if (proxyServer.isHttpPipeliningBlocked()) {
+            pipeline.addLast("httpPipeliningBlocker", new HttpPipeliningBlocker());
+        }
+
+        if (proxyServer.getRequestTracer() != null) {
+            pipeline.addLast("requestTracerHandler", new RequestTracerHandler(this));
```

Comment:
what's the impact of re-ordering these items?

Reply:
RequestTracerHandler.start() sets trace-id and removes thread-local values like tenant, preferences, etc. It is now activated when full http request is received instead of when a new group of bytes arrives. (Usually all http request bytes are sent all in once but if they are sent in several groups the handler might be called several times, which is incorrect)

From what I can see event logs and other important logs shouldn't be impacted by this change in a bad way.

Reply:
To sum up, RequestTracerHandler should be called per http request, not just per group of bytes